app/assets/javascripts/rentals/new.js.erb

```diff
@@ -13,40 +13,47 @@ function days_between(sd, ed) {
   return Math.round(diff_ms/one_day) + 1;
 }
 
+function item_types_to_arr() {
+  //collect item types (will always be at least [])
+  return $(".item-type-quantity-picker").toArray().reduce(function (acc,part) {
```

Comment:
sorry the explanation is on the method that calls this one, `app/assets/javascripts/new.js.erb:84`

Reply:
I read that. I'm still not clear on why things are being pushed a *dynamic* number of times.

Reply:
because of the way im constructing the parameters to `Rental.new` I need rentals_items_attributes in a way such that multiple item types yield multiple rentals_items not just a single one.  I could have this logic here or in the controller, I just chose to have it client side.